src/util/handlers/ParallelHandler.ts

```diff
@@ -0,0 +1,23 @@
+import { AsyncHandler } from './AsyncHandler';
+
+/**
+ * A composite handler that executes handlers in parallel.
+ */
+export class ParallelHandler<TIn = void, TOut = void> extends AsyncHandler<TIn, TOut[]> {
+  private readonly handlers: AsyncHandler<TIn, TOut>[];
+
+  public constructor(handlers: AsyncHandler<TIn, TOut>[]) {
+    super();
+    this.handlers = [ ...handlers ];
+  }
+
+  public async canHandle(input: TIn): Promise<void> {
+    // eslint-disable-next-line @typescript-eslint/promise-function-async
+    await Promise.all(this.handlers.map((handler): Promise<void> => handler.canHandle(input)));
+  }
```

Comment:
Is there a reason to differ from SequenceHandler here, as that one will always succeed and only executes the handlers that do support the input? Not that one way of handling it feels specifically better than the other, but we might want to be consistent there.

Reply:
`SequenceHandler` was really meant for "trying", I presume. Just going through a list and execute those that apply.

`ParallelHandler` is: I need all of those, and they can even run in parallel.
Makes the return value also easier.

No very strong opinion but simplicity.